src/security/PeerOptions.cc

```diff
@@ -62,6 +62,13 @@ Security::PeerOptions::parse(const char *token)
         parseOptions();
     } else if (strncmp(token, "cipher=", 7) == 0) {
         sslCipher = SBuf(token + 7);
+#if USE_OPENSSL
+        // everything okay
+#elif USE_GNUTLS
+        debugs(3, DBG_PARSE_NOTE(1), "WARNING: SSL cipher= not supported. Uses GnuTLS tls-options= parameters instead.");
+#else
+        debugs(3, DBG_PARSE_NOTE(1), "WARNING: SSL cipher= option requires --with-openssl.");
```

Comment:
Why not throw a configuration error?

Reply:
Because this is part of a complex line with many options that need different validation. The exception handling in Squid does not yet cope with this type of config line and the "-k parse" need to display more than one error on a single config line.

Reply:
FTR: This change request is very similar to another [one](https://github.com/squid-cache/squid/pull/139#discussion_r169866115). We can postpone resolving this change request until that other other is resolved.

Reply:
Closed as a duplicate of that other discussion thread about the GnuTLS clause immediately above. The resolution for both clauses will be the same.

Reply:
Unclosed to keep it as a marker for the requested changes. The discussion can indeed happen elsewhere.